packages/react/src/components/disclosure/disclosure.tsx

```diff
@@ -0,0 +1,68 @@
+import { FunctionComponent, PropsWithChildren, useState, useCallback } from 'react';
+import styled from 'styled-components';
+import { useId } from '../../hooks/use-id';
+import { Button, ButtonProps } from '../buttons/button';
+import { useDeviceContext } from '../device-context-provider/device-context-provider';
+
+type ButtonPropsWithoutOnClick = Omit<ButtonProps, 'onClick'>;
+
+interface DisclosureWidgetProps {
+    idContent?: string;
+    buttonProps: ButtonPropsWithoutOnClick;
+}
+
+export const Container = styled.div<{ $expanded: boolean; }>`
+    background-color: ${({ theme }) => theme.component['dropdown-menu-background-color']};
+    border:
+        ${({
+            $expanded,
+            theme,
+        }) => (
+            $expanded ? `1px solid ${theme.component['dropdown-menu-border-color']}` : 0
+        )};
+    border-radius: var(--border-radius);
+    box-shadow: 0 10px 20px 0 ${({ theme }) => theme.component['dropdown-menu-box-shadow-color']};
+    color: ${({ theme }) => theme.component['dropdown-menu-text-color']};
+    list-style-type: none;
+    position: absolute;
+    width: 100%;
+    z-index: 700;
+`;
+
+export const StyledButton = styled(Button)<{ isMobile: boolean }>`
+    font-size: 0.875rem;
+    font-weight: var(--font-normal);
+    text-transform: unset;
+`;
+
+const DisclosureContainer = styled.div`
+    position: relative;
+`;
+
+export const Disclosure: FunctionComponent<PropsWithChildren<DisclosureWidgetProps>> = ({
+    idContent: providedIdContent,
+    buttonProps,
+    children,
+}) => {
+    const [expanded, setExpanded] = useState<boolean>(false);
+    const idContent = useId(providedIdContent);
+    const { isMobile } = useDeviceContext();
+    const handleOnButtonClick = useCallback(() => {
+        setExpanded(!expanded);
+    }, [expanded, setExpanded]);
+
+    return (
+        <DisclosureContainer>
+            <StyledButton
+                {...buttonProps /* eslint-disable-line react/jsx-props-no-spreading */}
+                isMobile={isMobile}
+                onClick={handleOnButtonClick}
+                aria-expanded={expanded}
+                aria-controls={idContent}
+            />
+            <Container $expanded={expanded} id={idContent}>
+                {expanded && children}
```

Comment:
C'est weird de garder un container vide quand on est pas expanded. Après un peu de lecture sur `aria-controls`, on aurait 2 choix:
- Retirer le container et retirer la prop `aria-controls`. Selon MDN:
  > The aria-controls only needs to be set when the popup is visible, but it is valid and easier to program to reference an element that is not visible.
- Conserver la prop aria en tout temps mais conserver aussi le contenu du disclosure dans le DOM. C'est ce qui est mis de l'avant dans plusieurs exemples sur w3.org/WAI/ARIA

PS: Dans nos dropdown list/combobox, le comportement n'est pas bon car on retire complètement le container tout en gardant la prop aria-controls, ce qui ne semble pas être être correct

@LarryMatte ?

Reply:
en ce moment, on render pas le children quand expanded est à false, mais le container qui a le id du aria-controls est encore là (il est juste vide). est-ce que c'est vraiment nécessaire de render le contenu du div en tout temps ? dans le fond, tu mettrais la hauteur à zero quand expanded est à false au lieu de rien mettre dans le contenu ?

Reply:
je pense que je préfère garder ça dans le dom et de juste cacher la div dans le css.

Reply:
On pourrait effectivement mettre un `display: none;` sur le `Container` lorsqu'il est collapsed.

Reply:
Ouais, je parlais d'un `display: none`. Exemple: https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/